### PR TITLE
Besu skeleton

### PR DESCRIPTION
**Pull request by dragan2234**
I've made a PR for you branch @thomas-quadratic  so we can put comments from @matkt

### CODE REVIEW

File: ipa-multipoint/src/main/java/org/hyperledger/besu/ethereum/trie/verkle/BranchNode.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+public class BranchNode<V> implements Node<V> {
+    @SuppressWarnings("rawtypes")
+    protected static final Node NULL_NODE = NullNode.instance();
+
+    private final Optional<Bytes> location;  // Location in the tree
+    private final Bytes path;  // common children's trie-key prefix
+    private Bytes32 hash;  // vector commitment of all children's commitments
+    private boolean isDirty = false;  // should be flushed to storage
+    private boolean needsHealing = false;  // should children be synced with storage
```
Comment (dragan2234) [inline ipa-multipoint/src/main/java/org/hyperledger/besu/ethereum/trie/verkle/BranchNode.java:19]: @matkt comment:
"This healing variable is for snapsync. I think we don't have a snapsync mechanism for verkle trie yet so for me we can remove this one"

File: ipa-multipoint/src/main/java/org/hyperledger/besu/ethereum/trie/verkle/BranchNode.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    }
+
+    @Override
+    public String print() {
+        // TODO Auto-generated method stub
+        throw new UnsupportedOperationException("Unimplemented method 'print'");
+    }
+
+    @Override
+    public boolean isHealNeeded() {
```
Comment (dragan2234) [inline ipa-multipoint/src/main/java/org/hyperledger/besu/ethereum/trie/verkle/BranchNode.java:120]: @matkt comment:
"This healing variable is for snapsync. I think we don't have a snapsync mechanism for verkle trie yet so for me we can remove this one"

File: ipa-multipoint/src/main/java/org/hyperledger/besu/ethereum/trie/verkle/BranchNode.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        // TODO Auto-generated method stub
+        throw new UnsupportedOperationException("Unimplemented method 'isHealNeeded'");
+    }
+
+    @Override
+    public void markHealNeeded() {
```
Comment (dragan2234) [inline ipa-multipoint/src/main/java/org/hyperledger/besu/ethereum/trie/verkle/BranchNode.java:126]: @matkt comment:
"This healing variable is for snapsync. I think we don't have a snapsync mechanism for verkle trie yet so for me we can remove this one"

**Comment by thomas-quadratic** [comment]
I have resolved the above comments about snapsync.
Also got getVisitor and putVisitor going, with updated tests.
Seems ready to merge to me, it is already an overloaded PR.

**Comment by dragan2234** [comment]
I can't approve the PR via github since I created a PR, so approving via comment. Great work!

**Pull request merged**